Meta/Lagom/Tools/CodeGenerators/JSSpecCompiler/DiagnosticEngine.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "DiagnosticEngine.h"
+
+namespace JSSpecCompiler {
+
+thread_local DiagnosticEngine* DiagnosticEngine::m_instance = nullptr;
```

Comment:
Does this really need to be thread_local? Or are you intending to multi-thread this thing and have different diagnostics per thread?